vsts/node-release.yaml

```diff
@@ -0,0 +1,185 @@
+      STORAGE_CONNECTION_STRING: $(WINDOWS-STORAGE-CONNECTION-STRING)
+      IOT_PROVISIONING_DEVICE_IDSCOPE: $(WINDOWS-IOT-PROVISIONING-DEVICE-IDSCOPE)
+      IOT_PROVISIONING_ROOT_CERT: $(WINDOWS-IOT-PROVISIONING-ROOT-CERT)
+      IOT_PROVISIONING_ROOT_CERT_KEY: $(WINDOWS-IOT-PROVISIONING-ROOT-CERT-KEY)
+      IOT_PROVISIONING_SERVICE_CONNECTION_STRING: $(WINDOWS-IOT-PROVISIONING-SERVICE-CONNECTION-STRING)
+      DPS_CONN_STRING_INVALID_CERT: $(DPS-CONN-STRING-INVALID-CERT)
+      DPS_GLOBAL_DEVICE_ENDPOINT_INVALID_CERT: $(DPS-GLOBAL-DEVICE-ENDPOINT-INVALID-CERT)
+      IOTHUB_CONN_STRING_INVALID_CERT: $(IOTHUB-CONN-STRING-INVALID-CERT)
+      IOTHUB_DEVICE_CONN_STRING_INVALID_CERT: $(IOTHUB-DEVICE-CONN-STRING-INVALID-CERT)
+    enabled: false
+
+  - task: PublishTestResults@2
+    displayName: 'Publish Test Results | Mocha'
+    inputs:
+      testResultsFiles: 'e2etests/test-results.*.xml'
+      mergeTestResults: true
+      testRunTitle: 'E2E Tests - Windows'
+    condition: succeededOrFailed()
+    enabled: false
+
+  - powershell: |
+      $folders = @(
+        'common/core',
+        'common/transport/amqp',
+        'common/transport/http',
+        'common/transport/mqtt',
+        'device/core',
+        'device/transport/amqp',
+        'device/transport/http',
+        'device/transport/mqtt',
+        'device/node-red',
+        'service',
+        'provisioning/service',
+        'provisioning/device',
+        'provisioning/transport/amqp',
+        'provisioning/transport/mqtt',
+        'provisioning/transport/http',
+        'security/x509',
+        'security/tpm',
+        'security/symmetric')
+
+      $repoParent = $(Get-Item "$(Agent.BuildDirectory)").Parent
+
+      $artifacts = '$(Agent.RootDirectory)\.tar'  # artifacts folder path 
+
+      if ($(Test-Path $artifacts -PathType Container) -eq $true) {
+          Remove-Item -Path $artifacts -Recurse  # delete old artifacts folder
+      }
+
+      New-Item -ItemType Directory -Path $artifacts  # make new artifacts folder
+
+      foreach ($folder in $folders) {
+
+        $path = Join-Path -Path "$(Agent.BuildDirectory)\s" -ChildPath $folder  # archive file project root folder
+        
+        if ($(Test-Path -Path $path -PathType Container) -ne $true) {
+            throw "Folder not found: $($path)"
+        }
+        
+        $packageFile = Join-Path -Path $path -ChildPath 'package.json'  # archive file manifest
+            
+        if ($(Test-Path -Path $packageFile -PathType Leaf) -ne $true) {
+            throw "Package not found: $($packageFile)"
+        }
+        
+        Set-Location -Path $path  # set working folder to project root
+        
+        $json = Get-Content -Raw -Path $packageFile | ConvertFrom-Json  # read package.json to find output file name
+        $archiveName = "$($json.name)-$($json.version).tgz"
+        $archivePath = Join-Path -Path $path -ChildPath $archiveName
```

Comment:
note for later: we should be able to extract the archive name from the output of the NPM command rather than recreate it and hope they never change how they name packages.